File: 165. Compare Version Numbers.go
func compareVersion(version1 string, version2 string) int {
	revision1 := strings.Split(version1, ".")
	revision2 := strings.Split(version2, ".")
	for i := 0; i < len(revision1) || i < len(revision2); i++ {
		var num1, num2 int
		if i < len(revision1) {
			num1, _ = strconv.Atoi(revision1[i])
		}
		if i < len(revision2) {
			num2, _ = strconv.Atoi(revision2[i])
		}
		if num1 > num2 {
			return 1
		}
		if num1 < num2 {
			return -1
		}
	}
	return 0
}

File: 165. Compare Version Numbers.java
class Solution {
    public int compareVersion(String version1, String version2) {
        String[] revision1 = version1.split("\\.");
        String[] revision2 = version2.split("\\.");
        int i = 0, j = 0;
        while (i < revision1.length || j < revision2.length) {
            int r1 = i < revision1.length ? Integer.parseInt(revision1[i]) : 0;
            int r2 = i < revision2.length ? Integer.parseInt(revision2[i]) : 0;
            if (r1 > r2) return 1;
            else if (r1 < r2) return -1;
            i++;
            j++;
        }
        return 0;
    }
}